hooks/useCreateCollection.ts

```diff
@@ -61,25 +62,41 @@ export const useCreateCollection = (): useCreateCollectionReturn => {
   const { mutate: createCollectionMutation, isLoading: isCreatingCollection } =
     useMutation(
       (createServicesListRequests: CreateServicesListRequest[]) =>
-        createServicesLists(createServicesListRequests),
+        postServicesList(createServicesListRequests),
       {
         onSuccess: (data) => {
           router.push(`/list/${data[0].id}`)
         },
       }
     )
 
+  const userInfo = useUser()
   const onValidSubmit: SubmitHandler<CreateCollectionForm> = (data) => {
-    createCollectionMutation([
-      {
-        // Set status to "Draft" b/c research team has to approve new collections
-        status: 'Draft',
-        name: data.name,
-        description: data.description,
-        creator: data.creator,
-        resources: data.resources.map((resource) => resource.id),
-      },
-    ])
+    // This should be handled somewhere else (i.e. shouldn't be able to submit unless there's userInfo)
+    if (userInfo.user === null) {
+      createCollectionMutation([
+        {
+          // Set status to "Draft" b/c research team has to approve new collections
+          status: 'Draft',
+          name: data.name,
+          description: data.description,
+          creator: data.creator,
+          resources: data.resources.map((resource) => resource.id),
+        },
+      ])
+    } else {
+      createCollectionMutation([
+        {
+          // Set status to "Draft" b/c research team has to approve new collections
+          status: 'Draft',
+          name: data.name,
+          description: data.description,
+          creator: data.creator,
+          resources: data.resources.map((resource) => resource.id),
+          userId: userInfo.user.uid,
+        },
+      ])
+    }
```

Comment:
Not sure if this is the best way to do but my plan was to have the UI prompt the user to login before they could submit so this endpoint would never be hit without a user being logged in.

A better way to do it could be to just submit `userInfo.user.uid` regardless of whether or not it's defined and check it in the endpoint and maybe throw an error there. Either way, we can get rid of this conditional.

Reply:
The most like correct way to do this would prob be to add some Firebase auth token from when users are logged in to the request headers and check that in the backend but we can prob save that for a future MR.